modules/private/class-jetpack-private.php

```diff
@@ -0,0 +1,329 @@
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php
+			/* translators: URL for Jetpack dashboard. */
+			wp_kses( sprintf( __( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), admin_url( 'admin.php?page=jetpack' ) . '#/security?term=private' ), array( 'a' => array( 'href' => true ) ) );
+			?>
+		</div>
+		<fieldset>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	public static function private_robots_txt( $output ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+		return $output;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	public static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax and post requests on private blogs for users who don't have permissions
+	 */
+	public static function private_blog_prevent_requests() {
+		global $pagenow, $wpdb;
+
+		$this_current_user     = wp_get_current_user();
+		$is_ajax_request       = defined( 'DOING_AJAX' ) && DOING_AJAX;
+		$is_admin_post_request = ( 'admin-post.php' === $pagenow );
+
+		if ( is_super_admin() ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! $is_ajax_request && ! $is_admin_post_request ) ) {
+			return;
+		}
+
+		if ( ! self::is_private_blog_user( $wpdb->blogid, $this_current_user ) ) {
+			wp_die( '', 403 );
+		}
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	public static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $wpdb;
+
+		$this_current_user = wp_get_current_user();
+
+		if ( is_super_admin() ) {
+			return;
+		}
+
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		// These two ajax actions relate to wp_ajax_wp_link_ajax() and wp_ajax_find_posts()
+		// They are needed for users with admin capabilities in in wp-admin.
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! self::is_private_blog_user( $wpdb->blogid, $this_current_user ) ) {
+			wp_die( '', 403 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	public static function disable_rest_api() {
```

Comment:
We could hook later onto `rest_authentication_errors`. The docs indicate that this is the expected hook for restricting API access.

The cookie auth check happens at priority 100, so could `add_filter('rest_authentication_errors', 'disable_rest_api', 999)` or whatnot to stay in line with core's expectations. The current hook is probably okay too.

Reply:
> The current hook is probably okay too.

I'm retaining the current hook, unless you feel strongly about using `rest_authentication_errors`.